pkg/platform/local/client/store.go

```diff
@@ -370,19 +371,33 @@ func (s *Store) getResources(resourceDir string,
 func (s *Store) writeFileContents(filePath string, contents []byte) error {
 	s.logger.DebugWith("Writing file contents", "path", filePath, "contents", string(contents))
 
-	// get the file dir
-	fileDir := path.Dir(filePath)
+	tempFile, err := os.CreateTemp(".", "nuclio-contents-temp-file-*")
+	if err != nil {
+		return errors.Wrap(err, "Error creating temporary file")
+	}
 
-	// set NUCLIO_CONTENTS as base64 encoded value
-	env := map[string]string{"NUCLIO_CONTENTS": base64.StdEncoding.EncodeToString(contents)}
+	// remove the temporary file at the end
+	defer os.Remove(tempFile.Name())
 
-	// generate a command
-	_, _, err := s.runCommand(env,
-		`/bin/sh -c "mkdir -p %s && /bin/printenv NUCLIO_CONTENTS > %s"`,
-		fileDir,
-		filePath)
+	// encode contents to base64 and add a newline at the end
+	// newline at the end is needed to be able to parse files one be one
+	// when doing `cat /functions/*`
+	encodedContents := base64.StdEncoding.EncodeToString(contents) + "\n"
+
+	// write content to the temporary file
+	if _, err = tempFile.WriteString(encodedContents); err != nil {
+		tempFile.Close()
```

Comment:
don't you need `nolint: errcheck` here?

Reply:
added, but here link was OK even without it which is weird btw